modules/gui/src/com/haulmont/addon/dashboard/gui/components/DashboardFrame.java

```diff
@@ -33,26 +33,85 @@
 
     String NAME = "dashboard";
 
+    /**
+     * Initialize dashboard with passed parameters, timer and assistant.
+     *
+     * @param params map of dashboard params
+     */
+    void init(Map<String, Object> params);
+
+    Dashboard getDashboard();
+
+    /**
+     * Returns widget by passed id
+     *
+     * @param widgetId widget identifier
+     */
+    ScreenFragment getWidget(String widgetId);
+
+    /**
+     * Refreshes dashboard.
+     * Dashboard will be refreshed .
+     * If existed parameter has the same name as one of the param from passed map, it will be overwritten by param from map.
+     */
+    void refresh();
+
+    /**
+     * Refreshes dashboard with passed parameters.
+     * Dashboard will be refreshed with merged existed and new parameters.
+     * If existed parameter has the same name as one of the param from passed map, it will be overwritten by param from map.
+     *
+     * @param params map with new dashboard parameters
+     */
+    void refresh(Map<String, Object> params);
+
+    /**
+     * Dashboard can be configured by setting code of existing dashboard
+     *
```

Comment:
please add a note that `init` method required to apply new dashboard by code

Reply:
updated